src/targetPacks/assemble.targets

```
@@ -1,9 +1,81 @@
+          $(MSBuildProjectDirectory)\**\System.Activities.Presentation.il;
+          $(MSBuildProjectDirectory)\**\System.ComponentModel.Composition.il;
+          $(MSBuildProjectDirectory)\**\System.Data.Common.il;
+          $(MSBuildProjectDirectory)\**\System.Data.Entity.il;
+          $(MSBuildProjectDirectory)\**\System.Data.il;
+          $(MSBuildProjectDirectory)\**\System.Design.il;
+          $(MSBuildProjectDirectory)\**\System.Reflection.Metadata.il;
+          $(MSBuildProjectDirectory)\**\System.Web.il;
+          $(MSBuildProjectDirectory)\**\System.Xml.il;
+          $(MSBuildProjectDirectory)\**\System.XML.il;
+          $(MSBuildProjectDirectory)\**\WindowsFormsIntegration.il" />
+
+      <!-- Find all IL. Leave out IL that needs to be patched before being built. -->
+      <TargetingPackSrcNotRequiringConstraintPatch
+        Include="$(MSBuildProjectDirectory)\**\*.il"
+        Exclude="@(TargetingPackSrcRequiringConstraintPatch)" />
   </ItemGroup>
-  <Target Name="AssembleTargetPack" Condition="'@(TargetingPackSrc)' != '' and '$(SkipTargetingPacks)' != 'true' ">
+
+  <Target Name="GetProjectSrc"
+          BeforeTargets="PatchIl">
+    <PropertyGroup>
+      <PatchedILSrcDir>$(RepoRoot)artifacts/patchedILSrc/</PatchedILSrcDir>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <TargetingPackSrcNotRequiringConstraintPatch
+        ILSrcFile="%(FullPath)"
+        RelativeOutputAssemblyFile="%(RecursiveDir)%(Filename).dll" />
+      <TargetingPackSrcRequiringConstraintPatch
+        ILSrcFile="$(PatchedILSrcDir)$(MSBuildProjectName)/%(RecursiveDir)%(Filename).patched%(Extension)"
+        RelativeOutputAssemblyFile="%(RecursiveDir)%(Filename).dll"
+        DestinationFile="$(PatchedILSrcDir)$(MSBuildProjectName)/%(RecursiveDir)%(Filename).patched%(Extension)" />
+      <TargetingPackSrc
+        Include="@(TargetingPackSrcNotRequiringConstraintPatch);@(TargetingPackSrcRequiringConstraintPatch)" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="PatchIl"
+          BeforeTargets="AssembleTargetPack"
+          Condition="'@(TargetingPackSrcRequiringConstraintPatch)' != ''">
+
+    <!--
+      Wrap constraint in single quotes in certain circumstances to avoid using reserved token.
+      Function param:
+        " constraint," => " 'constraint',"
+        " constraint)" => " 'constraint')"
+        " constraint"(eol) => " 'constraint'"(eol) (%24 is $ escaped in hex)
+    -->
+    <Exec Command="
+      set -e
+      destination=&quot;%(TargetingPackSrcRequiringConstraintPatch.DestinationFile)&quot;
+      mkdir -p &quot;${destination%/*}&quot;
+      sed -E &quot;s/ constraint(([,)])|%24)/ 'constraint'\2/g&quot; &quot;%(TargetingPackSrcRequiringConstraintPatch.FullPath)&quot; > &quot;$destination&quot;" />
+
+  </Target>
+
+  <Target Name="AssembleTargetPack"
+          Condition="'@(TargetingPackSrcNotRequiringConstraintPatch)' != '' and '$(SkipTargetingPacks)' != 'true' ">
```

Shouldn't this still run even if all IL files require the constraint patch?

```suggestion
          Condition="'@(TargetingPackSrc)' != '' and '$(SkipTargetingPacks)' != 'true' ">
```

---

Yes.  I fixed this. Thanks.